### PR TITLE
New batching library for OpenVINO model inference Benchmark App tests

### PR DESCRIPTION
**Pull request by tsai-henry**
The Batch class is a library for engineers to automate the running and logging of [OpenVINO Deep Learning Inference benchmark tests](https://docs.openvino.ai/2023.0/openvino_inference_engine_tools_benchmark_tool_README.html). To use the Batch class, please follow these four simple steps:

1. pip install -r requirements.txt
2. Set the environment variable OPENVINO_PATH to the folder where [OpenVINO](https://docs.openvino.ai/2023.0/openvino_docs_install_guides_overview.html) is installed on your machine. On Linux machines this will usually be "/opt/intel/openvino_2022" but with your specific version. This is necessary for initializing the OpenVINO environment, otherwise all benchmarks will fail.
3. python3 main.py [MODEL_FOLDER] [CONFIG_FILE]
Replace MODEL_FOLDER with the path of your machine's OpenVINO model folder and CONFIG_FILE with the path to a JSON file containing your benchmark settings. On Linux machines the model folders will usually be "~/public" or "~/intel". You can find examples for both in this repository in the example_model folder and example_settings.json file. If these two arguments are omitted in the command line, it will automatically select the example paths.
4. Open the output folder to view a timestamped subfolder containing all benchmark results, including full text logs and a summary CSV.

This library was developed by Henry Tsai, Software Engineering Intern at Intel Automotive Chandler, in July 2023 under the guidance of Rony Ferzli and Keith Perrin. The original usage for the library was to benchmark face detection models for automotive applications. As such, there is a Jupyter Notebook included containing data analysis of selected face detection model results which can be accessed with the terminal command jupyter notebook and opening the resulting window.

Future development of this library could include:
Adding support for additional benchmark app arguments in the benchmark method
Adding support for benchmarking .onnx model files (currently only supports native OpenVINO .xml files)
Adding functionality for automatically converting non-native model files before running benchmarks using the [OpenVINO Model Converter](https://docs.openvino.ai/2023.0/omz_tools_downloader.html#model-converter-usage)
Improving error handling for CSV summary generation on error logs
Including system information (e.g. Linux lscpu/clinfo commands) in logs

### CODE REVIEW

**Comment by ilya-lavrenov** [comment]
Please, don't push models to repository.

**Comment by tsai-henry** [comment]
@DariaMityagina please let me know if you need any additional information, my internship and intel account access will be ending today

File: tests/.gitignore [hunk 1]
```diff
@@ -0,0 +1,162 @@
+output/*
```
Comment (DariaMityagina) [inline tests/.gitignore:1]: I'm not sure that `tests` is the right place for your tool.
Could you please consider moving it to the `tools` folder?

File: tests/Batch/output/all_openvino_models_i7-1370PRE_CPU_iGPU/bert-large-uncased-whole-word-masking-squad-int8-0001_08_25_2023-16_12_19_log.txt [hunk 2]
```diff
@@ -0,0 +1,312 @@
+Error benchmarking model
```
Comment (DariaMityagina) [inline tests/Batch/output/all_openvino_models_i7-1370PRE_CPU_iGPU/bert-large-uncased-whole-word-masking-squad-int8-0001_08_25_2023-16_12_19_log.txt:1]: `Output` folder contains too many large files that are not really needed for the tool to work. 
You can leave a small example or describe everything in the `README`.
Could you please remove unnecessary logs from the PR and leave only needed files (with the source code)?

Comment (DariaMityagina) [reply tests/Batch/output/all_openvino_models_i7-1370PRE_CPU_iGPU/bert-large-uncased-whole-word-masking-squad-int8-0001_08_25_2023-16_12_19_log.txt:1]: `Batch/*.ipynb` analysis files are quite representative. I believe it should be okay to keep them.
If OpenVINO maintainers do not mind.

**Pull request closed without merging**